caom2-artifact-sync/src/test/java/ca/nrc/cadc/caom2/artifactsync/ArtifactMetadataTest.java

```
@@ -171,16 +187,71 @@ private void testCompareMetadata(boolean reportOnly) throws Exception {
         logicalMetadata = new ArtifactMetadata();
         physicalMetadata = new ArtifactMetadata();
         logicalMetadata.storageID = "bar";
-        physicalMetadata.artifactURI = "mast:HST/product/id5n04lfq_drc.fits";
+        logicalMetadata.artifactURI = new URI("mast:HST/product/id5n04lfq_drc.fits");
+        logicalMetadata.productType = ProductType.SCIENCE;
         physicalMetadata.checksum = "1043fe4c1a259a610fa9fb7ebff5833f";
         physicalMetadata.contentLength = "10";
         physicalMetadata.contentType = "logicalType";
-        physicalMetadata.collection = "HST";
         physicalMetadata.lastModified = new Date();
-        physicalMetadata.releaseDate = new Date();
         physicalMetadata.storageID = "bar";
         logicalArtifacts.add(logicalMetadata);
         physicalArtifacts.add(physicalMetadata);
         validator.compareMetadata(logicalArtifacts, physicalArtifacts, start);
     }
+    
+    private class CADCArtifactStore implements ArtifactStore {
```

If this test impl is needed to test and react to the containsPublicOnly then I'd name it something other than CADCArtifactStore (just TestArctifactStore is sufficient) and I would not bother with making it configurable by file.

Just have a ctor like TestArtifactStore(boolean) and create different instances for different tests. 

Methods you do not expect tests to call should throw an exception if called; UnsupportedOperationException seems approp.